src/include/control_points.h

```diff
@@ -31,25 +31,36 @@ typedef uint32_t wt_control_point_action_id_t;
  * A function to initialize a control point's data.
  * If per-connection session = NULL.
  */
-typedef WT_CONTROL_POINT_DATA *wt_control_point_init_t(
-  WT_SESSION_IMPL *session, const char *cp_config_name, const char **cfg);
+typedef WT_CONTROL_POINT_DATA *wt_control_point_init_t(WT_SESSION_IMPL *session,
+  const char *cp_config_name, bool control_point_for_connection,
+  int (*pred_func)(WT_SESSION_IMPL *, WT_CONTROL_POINT_DATA *, WT_CONFIG_ITEM *), const char **cfg);
 
 /*!
  * A function to test whether a control point should be triggered.
  */
 typedef bool wt_control_point_pred_t(
   WT_SESSION_IMPL *session, WT_CONTROL_POINT_REGISTRY *cp_registry, WT_CONTROL_POINT_DATA *data);
 
+/*!
+ * A function to initialize predicate.
+ */
+typedef int wt_control_point_init_pred_t(
+  WT_SESSION_IMPL *session, WT_CONTROL_POINT_DATA *data, WT_CONFIG_ITEM *item);
+
 /*!
  * Registration data for one control point.
  */
 struct __wt_control_point_registry {
     wt_control_point_init_t __F(init); /* Function to initialize the control point. */
+    wt_control_point_init_pred_t __F(
+      init_pred);                      /* Function to initialize the predicate control point. */
     wt_control_point_pred_t __F(pred); /* Function to test whether to trigger. */
     size_t crossing_count;             /* Count of executions of the trigger site. */
     size_t trigger_count;              /* Count of triggers, i.e. pred returned true. */
     WT_SPINLOCK lock;                  /* Atomically access data and data->ref_count. */
     const char *config_name;           /* Control point config name */
+    bool enable_at_open;
+    bool control_point_for_connection;
```

Comment:
Delete line 63. That is implied by whether it is stored in WT_CONNECTION.control_points or WT_SESSION.control_points.